cli/support_test.go

```diff
@@ -156,6 +160,42 @@ func TestSupportBundle(t *testing.T) {
 		err := inv.Run()
 		require.ErrorContains(t, err, "failed authorization check")
 	})
+
+	// This ensures that the CLI does not panic when trying to generate a support bundle
+	// against a fake server that returns a 200 OK for all requests. This essentially
+	// ensures that (almost) all of the support bundle generating code paths get a zero value.
+	t.Run("DontPanic", func(t *testing.T) {
+		t.Parallel()
+
+		// Start up a fake server that will return a blank 200 OK response for everything.
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			t.Logf("received request: %s %s", r.Method, r.URL)
+			switch r.URL.Path {
+			case "/api/v2/authcheck":
+				// Fake auth check
+				resp := codersdk.AuthorizationResponse{
+					"Read DeploymentValues": true,
+				}
+				w.WriteHeader(http.StatusOK)
+				assert.NoError(t, json.NewEncoder(w).Encode(resp))
+			default:
+				// Simply return a 200 OK for everything else.
+				w.WriteHeader(http.StatusOK)
```

Comment:
Would a non-200 response also trigger this problem?

For example:

```go
	eg.Go(func() error {
		hr, err := healthsdk.New(client).DebugHealth(ctx)
		if err != nil {
			return xerrors.Errorf("fetch health report: %w", err)
		}
		d.HealthReport = &hr
		return nil
	})
```

I think it might be more idiomatic to have all calls fail rather than return empty values, although both are worthwhile testing.

Reply:
That's a good point, I'll test against a range of status codes. :+1:

Reply:
Updated!